src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIGoHome.java

```diff
@@ -82,8 +117,84 @@ public void startExecuting()
         }
 
         playGoHomeSounds();
+        handleSaturation(pos);
+    }
+
+    /**
+     * Handle the saturation of the citizen.
+     *
+     * @param pos the position.
+     */
+    private void handleSaturation(@NotNull final BlockPos pos)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)